Mathlib/Order/Filter/AtTopBot.lean

```diff
@@ -576,6 +576,84 @@ theorem strictMono_subseq_of_id_le {u : ℕ → ℕ} (hu : ∀ n, n ≤ u n) :
 theorem _root_.StrictMono.tendsto_atTop {φ : ℕ → ℕ} (h : StrictMono φ) : Tendsto φ atTop atTop :=
   tendsto_atTop_mono h.id_le tendsto_id
 
+/-- If `f` is a monotone funciton and `g` tends to `atTop` along a nontrivial filter.
+then the upper bounds of the range of `f ∘ g`
+are the same as the upper bounds of the range of `f`.
+
+This lemma together with `exists_seq_monotone_tendsto_atTop_atTop` below
+is useful to reduce a statement
+about a monotone family indexed by a type with countably generated `atTop` (e.g., `ℝ`)
+to the case of a family indexed by natural numbers. -/
+theorem upperBounds_range_comp_of_monotone_of_tendsto_atTop [Preorder β] [Preorder γ]
+    {l : Filter α} [l.NeBot] {f : β → γ} (hf : Monotone f) {g : α → β} (hg : Tendsto g l atTop) :
+    upperBounds (range (f ∘ g)) = upperBounds (range f) := by
+  refine Subset.antisymm ?_ (upperBounds_mono_set <| range_comp_subset_range _ _)
+  rintro c hc _ ⟨b, rfl⟩
+  obtain ⟨a, ha⟩ : ∃ a, b ≤ g a := (hg.eventually_ge_atTop b).exists
+  exact (hf ha).trans <| hc <| mem_range_self _
+
+/-- If `f` is a monotone funciton with bounded range
+and `g` tends to `atTop` along a nontrivial filter,
+then the indexed supremum of `f ∘ g` is equal to the indexed supremum of `f`.
+
+See also the next two lemmas for the cases when the assumption `BddAbove (range f)` can be omitted.
+
+This lemma together with `exists_seq_monotone_tendsto_atTop_atTop` below
+is useful to reduce a statement
+about a monotone family indexed by a type with countably generated `atTop` (e.g., `ℝ`)
+to the case of a family indexed by natural numbers. -/
+theorem ciSup_monotone_comp_tendsto_atTop [Preorder β] [ConditionallyCompleteLattice γ]
```

Comment:
Do we need the `ConditionallyCompleteLattice` version? Should I add dual order versions?

Reply:
I think it's good to have the `ConditionallyCompleteLattice`, because it factors out nicely the other versions. And yes, could you please add the `antitone` versions?